utils/check_repo.py

```diff
@@ -467,32 +469,17 @@ def get_model_test_files():
     return test_files
 
 
-# This is a bit hacky but I didn't find a way to import the test_file as a module and read inside the tester class
-# for the all_model_classes variable.
 def find_tested_models(test_file):
-    """Parse the content of test_file to detect what's in all_model_classes"""
-    # This is a bit hacky but I didn't find a way to import the test_file as a module and read inside the class
-    with open(os.path.join(PATH_TO_TESTS, test_file), "r", encoding="utf-8", newline="\n") as f:
-        content = f.read()
-    all_models = re.findall(r"all_model_classes\s+=\s+\(\s*\(([^\)]*)\)", content)
-    # Check with one less parenthesis as well
-    all_models += re.findall(r"all_model_classes\s+=\s+\(([^\)]*)\)", content)
-    if len(all_models) > 0:
-        model_tested = []
-        for entry in all_models:
-            for line in entry.split(","):
-                name = line.strip()
-                if len(name) > 0:
-                    model_tested.append(name)
-        return model_tested
+    test_file = os.path.join(PATH_TO_TESTS, test_file)
+    return [x.__name__ for x in get_tested_models(test_file)]
```

Comment:
This is cleaner, but it requires we have all the dependency to be able to import the modules, as well as to have the complete `all_model_classes`.

For example, so far, we won't be able to have the model classes below (missing required libs)

```python
    all_model_classes = (
        (RagModel, RagTokenForGeneration, RagSequenceForGeneration)
        if is_torch_available() and is_datasets_available() and is_faiss_available()
        else ()
    )
```

Would like to know your thoughts before I continue.

Reply:
We don't really need to accept the change in this file. It's just a demonstration. But if you happen to have some ideas, let me know.

Reply:
I'd rather keep the old check then. For now `make repo-consistency` works without faiss installed for instance. This change would make it mandatory (plus all other exotic libs).

Reply:
No problem, ☺️